docs/autocomplete_compatibility.md

```diff
@@ -0,0 +1,41 @@
+# Autocomplete Compatability
+
+Many lists are used to drive autocompleting user interface components. In order to support this, a standardised concept of an "autocomplete compatible" list has been defined. A list which is autocomplete compatible has the information required to drive an autocomplete-based selection mechanism, and has that information in well-known fields so no additional configuration is required to use that list in an autocomplete mechanism that understands this standard.
+
+# Specification
+
+An autocomplete-compatible list must have:
+
+1. A `name` field, which is a string value in every record. This should be the "best" name for the identified object, such as a legal or official name.
+2. (Optionally) an `match_synonyms` field which, in every record in which it is present, is an array of strings. These should be alternative names by which the object may be known, but which can only refer to this object.
+2. (Optionally) a `suggestion_synonyms` field which, in every record in which it is present, is an array of strings. These should be alternative names by which the object may possibly be known, but which might not always refer to this object.
+2. (Optionally) an `hint` field which, in every record in which it is present, is a string. It should be a message to be shown to users to help them decide if this is the object they mean to select.
+
+In addition, for any given list, the combination of `name` and values inside `match_synonyms` must not contain duplicates - there can be no string that matches more than one record in the `name` field or a `match_synonym`. Duplication within `suggestion_synonyms` is fine, but no value inside `suggestion_synonyms` can also occur in a `name` or inside a `match_synonyms`.
```

Comment:
"but no value inside `suggestion_synonyms` can also occur in a `name` or inside a `match_synonyms`."

Unsure what this is trying to say. It might read that synonyms can't be strings that are a subset of the `name` - but I don't know why we'd not allow such an occurance.

Reply:
Ah yes, that was a mistake I thought I'd erased all traces of. Well spotted!